server.js

```diff
@@ -1,54 +1,74 @@
-var util = require('util');
-var url = require('url');
-var qs = require('querystring');
-var os = require('os')
+//  OpenShift sample Node application
+var express = require('express');
+var fs      = require('fs');
+var app     = express();
+var eps     = require('ejs');
+
+app.engine('html', require('ejs').renderFile);
+
 var port = process.env.PORT || process.env.port || process.env.OPENSHIFT_NODEJS_PORT || 8080;
-var ip = process.env.OPENSHIFT_NODEJS_IP || '0.0.0.0';
-var nodeEnv = process.env.NODE_ENV || 'unknown';
+var ip   = process.env.IP   || process.env.OPENSHIFT_NODEJS_IP || '0.0.0.0';
+var mongoURL = process.env.OPENSHIFT_MONGODB_DB_URL || process.env.MONGO_URL;
+if (mongoURL == null) {
+	var mongoPort = process.env.MONGODB_SERVICE_PORT || 27017;
+	var mongoDb = process.env.MONGODB_DATABASE || "pageviews";
```

Comment:
@sspeiche it'll be really hard to debug if you write a bad template, for example.

I feel your pain. I wonder if we should have a way to tell it that you're in a dev environment and be able to use a different configuration. Passing env var that tells the app to look for other settings. You could then just use one var for mongo url. Though I don't know if we have time for it right now.